src/EditorFeatures/Core/Implementation/IntelliSense/QuickInfo/IntellisenseQuickInfoBuilder.cs

```diff
@@ -79,16 +107,67 @@ internal static class IntellisenseQuickInfoBuilder
             }
 
             var content = new ContainerElement(
-                                ContainerElementStyle.Stacked,
+                                ContainerElementStyle.Stacked | ContainerElementStyle.VerticalPadding,
                                 elements);
 
             return new IntellisenseQuickInfoItem(trackingSpan, content);
         }
 
-        private static ClassifiedTextElement BuildClassifiedTextElement(QuickInfoSection section)
+        private static IEnumerable<object> BuildClassifiedTextElements(QuickInfoSection section)
         {
-            return new ClassifiedTextElement(section.TaggedParts.Select(
-                    part => new ClassifiedTextRun(part.Tag.ToClassificationTypeName(), part.Text)));
+            var paragraphs = new List<object>();
+            var currentParagraph = new List<object>();
+            var currentRuns = new List<ClassifiedTextRun>();
+            foreach (var part in section.TaggedParts)
+            {
+                if (part.Tag == TextTags.LineBreak)
+                {
+                    if (currentRuns.Count == 0)
+                    {
+                        if (currentParagraph.Count > 0)
+                        {
+                            if (currentParagraph.Count == 1)
+                            {
+                                paragraphs.Add(currentParagraph[0]);
+                            }
+                            else
+                            {
+                                paragraphs.Add(new ContainerElement(ContainerElementStyle.Stacked, currentParagraph));
+                            }
+
+                            currentParagraph.Clear();
+                        }
+                    }
+                    else
+                    {
+                        currentParagraph.Add(new ClassifiedTextElement(currentRuns));
+                        currentRuns.Clear();
+                    }
+                }
+                else
+                {
+                    currentRuns.Add(new ClassifiedTextRun(part.Tag.ToClassificationTypeName(), part.Text));
+                }
+            }
```

Comment:
so basically if there is multiple linebreaks then it got ignored.

Reply:
Currently yes. I can add tests verifying that each of the following produces the same result (the first being the preferred source form for this output):

1. `<para>text</para><para>text</para>`
2. `text<br/><br/>text`
3. `text<br/><br/><br/>text`
4. `<para>text</para><br/><para>text</para>`

Any objection to the behavior in 2-4 as described? If so, can you clarify how it should be presented?

Reply:
I don't have objection on what we decide on UX. confident that people who care on those were discussed and decided what the good experience is. I was more asking whether there is a test covering it so that people know the experience is intentional.

Reply:
I will add tests for each of the above.

Reply:
I have now _documented_ the ignored empty paragraphs, but not marking this resolved since I'm still working on the tests.

Reply:
➡️ Added documentation in 05ab07fee61a82a457e7c7952d15c5a08f4eb0b2, and added a bunch of tests in d8d897ee380395d72057d1bf4a9e3b91998e67d8. This suggestion caught a bug in the implementation. 👍